lib/tasks/tag_tasks.rake

```diff
@@ -36,16 +36,32 @@ namespace :Tag do
 
   desc "Delete unused tags"
   task(delete_unused: :environment) do
+    start = Time.current
     deleted_names = []
-    Tag.where(canonical: false, merger_id: nil, taggings_count_cache: 0).each do |t|
-      if t.taggings.count.zero? && t.child_taggings.count.zero? && t.set_taggings.count.zero?
+    Tag.where(canonical: false, merger_id: nil, taggings_count_cache: 0).find_in_batches do |batch|
+      batch.each do |t|
+        next unless t.taggings.count.zero?
+        next unless t.child_taggings.count.zero?
+        next unless t.set_taggings.count.zero?
+
         deleted_names << t.name
-        t.destroy
+        begin
+          t.destroy
+          print "+"
+        rescue ActiveRecord::LockWaitTimeout
+          sleep(10)
+          print("Retrying ", t.name)
+          retry
+        end
       end
+      print "."
     end
     unless deleted_names.blank?
       puts "The following #{deleted_names.length} unused tags were deleted:"
       puts deleted_names.join(", ")
+      puts "Started #{start}"
+      puts "Ended #{Time.current}"
+      puts "Deleted #{deleted_names.length}"
```

Comment:
I think we can remove this line. The length is already there a few lines up: `puts "The following #{deleted_names.length} unused tags were deleted:"`

Reply:
If it delete's thousands of tags it is really hard to find the start. The end of the log is really easy to find. I could delete the one at the top ?

Reply:
Yeah, that's fine!